[Peek][PreviewPane] Fix missing Copy menu-item

drawbyperpetual opened the pull request:
## Summary of the Pull Request
Fixes two bugs:
- Peek: Missing "Copy" menu-item for all WebView2 previewers.
- PreviewPane: Missing "Copy" menu-item for markdown files only.

## PR Checklist

- [X] **Closes:** #32449
- [X] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [ ] **Tests:** Added/updated and all pass
- [X] **Localization:** All end user facing strings can be localized
- [ ] **Dev docs:** Added/updated
- [ ] **New binaries:** Added on the required places
   - [ ] [JSON for signing](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ESRPSigning_core.json) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/main/installer/PowerToysSetup/Product.wxs) for new binaries and localization folder
   - [ ] [YML for CI pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ci/templates/build-powertoys-steps.yml) for new test projects
   - [ ] [YML for signed pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/release.yml)
- [ ] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/windows-uwp/tree/docs/hub/powertoys) and link it here: #xxx

## Detailed Description of the Pull Request / Additional comments
The issues are:
- Peek: 
  - When not using Monaco (markdown, html) - the default WebView2 context menu has been disabled. I have enabled it and then disabled ALL menu-items other than "Copy" (such as "Back").
  - When using Monaco + Release (other code files) - current code tries to use the Monaco context menu, but it is somehow disabled at runtime. I spent MANY hours trying to find out why but without success. It works fine when I view the generated html + js files in a browser or in a Debug build or in PreviewPane. But I couldn't find the root cause. Trying to fix it by enabling the WebView2 context menu instead doesn't work as for whatever reason, WebView2 doesn't generate a "Copy" menu-item (it thinks there's no selected text when there is). So in this case, the only thing I could get to work was generating context menu-items via WebView2 callbacks that call JS functions. As a bonus, this way of doing it also allows "Toggle text wrapping" to work.
- PreviewPane:
  - Markdown - the default WebView2 context menu has been disabled. Like for Peek, I have enabled it and then disabled ALL menu-items other than "Copy" (such as "Back").
  - Monaco (other code files) - this already just works fine, so I've left it as is. I *could* make it work the same way as I've done for Peek for consistency, but I've chosen to leave it as is since it works.
  
![image](https://github.com/user-attachments/assets/d758ada7-bb62-4f40-bef7-ad08ffb83786)
![image](https://github.com/user-attachments/assets/4e0baa7e-632f-412a-b2b1-b9f666277ca7)

## Validation Steps Performed
- Tested copy (via menu-item and CTRL+C) for text formats on Peek/PreviewPane. Also tested "Toggle text wrapping" where appropriate.
- Sanity check of Peek and PreviewPane.

drawbyperpetual committed:
[Peek] Made copy work on BrowserControl

drawbyperpetual committed:
Merge branch 'main' into dev/ani/peek-fix-copy

drawbyperpetual committed:
[Peek][PreviewPane] Fixed missing or disabled Copy menu item

drawbyperpetual committed:
Merge branch 'main' into dev/ani/peek-fix-copy

htcfreek commented:
@drawbyperpetual 
A screenshot of the new menu content would be great.

Not sure how it looks. But for the layout a separator line between the copy command and the setting toggle entries would be good, I think.

drawbyperpetual committed:
Added separator

drawbyperpetual committed:
Merge branch 'main' into dev/ani/peek-fix-copy

drawbyperpetual commented:
> @drawbyperpetual A screenshot of the new menu content would be great.
> 
> Not sure how it looks. But for the layout a separator line between the copy command and the setting toggle entries would be good, I think.

@htcfreek : I've added a separator between the commands and also updated the PR description with screenshots. Let me know what you think!

htcfreek commented:
> > @drawbyperpetual A screenshot of the new menu content would be great.
> > 
> > Not sure how it looks. But for the layout a separator line between the copy command and the setting toggle entries would be good, I think.
> 
> @htcfreek : I've added a separator between the commands and also updated the PR description with screenshots. Let me know what you think!

Looks good.

htcfreek commented:
Btw, merging this should be synced with merging #33742.

jaimecbernardo reviewed (changes requested):
Thanks for the feature.
LGTM. Only quirk I've found is that if I use Peek on either Markdown or Monaco files, make the context menu show and then move the Peek window between monitors, it will make the context menu start appearing on random spots. Like sometimes it even shows on a different monitor than what the Peek window is in.

drawbyperpetual committed:
Switched to deferred WPF menu to avoid strange menu location issues with WebView2 menu

drawbyperpetual commented:
> Thanks for the feature. LGTM. Only quirk I've found is that if I use Peek on either Markdown or Monaco files, make the context menu show and then move the Peek window between monitors, it will make the context menu start appearing on random spots. Like sometimes it even shows on a different monitor than what the Peek window is in.

@jaimecbernardo : I've fixed that context menu location issue for Peek. The way I did it was to not rely on the WebView2 context menu at all but instead draw a WPF context menu instead. The only downside is I couldn't find a way to replicate the pretty Copy icon from the WebView2 context menu. But I think that's a small issue as Monaco doesn't have it either.

BTW, the PreviewPane context menu doesn't seem to have to that  context menu location issue for whatever reason. So I've kept the WebView2 context menu for the PreviewPane + Markdown previewer instead of drawing a WPF context menu.

jaimecbernardo reviewed (approved):
LGTM! Thank you!
